synapse/cortex.py

```diff
@@ -198,6 +104,33 @@ def getCoreInfo(self):
                                           user=self.user)
         return iden
 
+    async def _reqViewAllowed(self, iden):
+        '''
+
+        Args:
+            iden(str): view iden to access
+
+        Returns:
+            None
+
+        Raises:
+            s_exc.AuthDeny: If the permission is not allowed
+            s_exc.NoSuchView: If the view iden doesn't exist
+
+        '''
+        if iden is None:
+            return self.cell.view
+
+        view = self.cell.views.get(iden)
+        if view is None:
+            raise s_exc.NoSuchView(iden=iden)
+
+        # TODO:  distinguish between read and read/write perms
+
+        await self._reqUserAllowed('view:lift', iden)
```

Comment:
Permission names probably bear thinking through a bit here.  Likely we'll want per-view permissions rather than "all views" permissions.

Reply:
specifically, this would only matter for non-default views. the default view is always accessible for access.

Reply:
removed